AirCasting/URLProviders/UserDefaultsBaseURLProvider.swift

```diff
@@ -3,7 +3,7 @@
 
 import Foundation
 
-class UserDefaultsBaseURLProvider: BaseURLProvider {
+class UserDefaultsBaseURLProvider: BaseURLProvider, ObservableObject {
```

Comment:
You use here ObservableObject but when declaring in RootAppView :` let urlProvider = UserDefaultsBaseURLProvider()` it is not wrapped in the @Enviroment or anything else.
For the time where everything depends on the userDefaults, does it need to conform to ObservableObject?

Reply:
To be passed as `EnvironmentObject` it needs to. I don't think this is a great approach to dependency management in SwiftUI (environment objects should be used to share data between views, not as a DI framework), but we have nothing better currently to work with

Reply:
@jankrzempek it's passed as  .`environmentObject(urlProvider)` and in `MainAppView` I use `@EnvironmentObject var urlProvider: UserDefaultsBaseURLProvider`